testutils/clustermanager/gke.go

```diff
@@ -260,11 +179,13 @@ func (gc *GKECluster) Provider() string {
 // in us-central1, and default BackupRegions are us-west1 and us-east1. If
 // Region or Zone is provided then there is no retries
 func (gc *GKECluster) Acquire() error {
+	// Make a deep copy of the request struct, since the original request is supposed to be immutable
+	request := gc.Request.DeepCopy()
```

Comment:
I would move this down to be just above where it's going to be used, since this code path doesn't need to be executed every time